test/AutoinstGeneral_test.rb

```diff
@@ -3,58 +3,141 @@
+        expect(Yast::SCR).not_to receive(:Execute)
+          .with(path(".target.bash"), /hwclock/)
+        subject.Write()
+      end
+
+      it "sets multipath" do
+        expect(Yast::AutoinstStorage).to receive(:set_multipathing)
+        subject.Write
+      end
+    end
+  end
 
-    it "syncing hardware time if ntp server is set" do
-      Yast::Profile.ReadXML(profile_sync)
-      Yast::AutoinstGeneral.Import(Yast::Profile.current["general"])
+  describe "#Import" do
+    context "when multipath is enabled" do
+      let(:profile) do
+        { "storage" => { "start_multipath" => true } }
+      end
 
-      expect(Yast::SCR).to receive(:Execute).with(
-        path(".target.bash"),
-        "/usr/sbin/ntpdate -b #{Yast::AutoinstGeneral.mode["ntp_sync_time_before_installation"]}").and_return(0)
-      expect(Yast::SCR).to receive(:Execute).with(
-        path(".target.bash"),"/sbin/hwclock --systohc").and_return(0)
+      it "sets multipath"
+    end
 
-      Yast::AutoinstGeneral.Write()
+    context "when multipath is not enabled" do
+      it "does not set multipath"
     end
 
-    it "not syncing hardware time if no ntp server is set" do
-      Yast::Profile.ReadXML(profile_no_sync)
-      Yast::AutoinstGeneral.Import(Yast::Profile.current["general"])
+    context "when btrfs default subvolume name is set" do
+      let(:profile) do
+        { "storage" => { "btrfs_set_default_subvolume_name" => "@@" } }
+      end
 
-      expect(Yast::SCR).not_to receive(:Execute).with(
-        path(".target.bash"),"/sbin/hwclock --systohc")
+      it "sets the default subvolume"
+    end
 
-      Yast::AutoinstGeneral.Write()
+    context "when btrfs default subvolume name is not set" do
+      it "uses the default name"
     end
 
+    context "when partitiong alignment is defined" do
+      let(:profile) do
+        { "storage" => { "partition_alignment" => "align_optimal" } }
+      end
+
+      it "sets partitiong alignment"
+    end
   end
 
+  describe "#Export" do
+    let(:profile) do
+      {
+        "storage"            => { "start_multipath" => true },
+        "mode"               => { "confirm" => false },
+        "signature-handling" => { "import_gpg_key" => true },
+        "ask-list"           => ["ask1"],
+        "proposals"          => ["proposal1"]
+
+      }
+    end
+
+    before do
+      subject.Import(profile)
+    end
+
+    it "exports storage settings" do
+      expect(subject.Export).to include("storage" => profile["storage"])
+    end
+
+    context "when the old 'btrfs_set_default_subvolume_name' is used" do
+      let(:profile) do
+        { "storage" => { "btrfs_set_default_subvolume_name" => "@" } }
+      end
+
+      it "exports that option renamed to 'btrfs_default_subvolume'" do
+        expect(subject.Export).to include("storage" => { "btrfs_default_subvolume" => "@" })
+      end
+    end
+
+    it "exports mode settings" do
+      expect(subject.Export).to include("mode" => profile["mode"])
```

Comment:
Shouldn't we also check the exported value (here an on all the equivalent ones below)?